### PR TITLE
pylint and packages fixes

### PR DESCRIPTION
**Pull request by chlorophyll-zz**
for ./check/all : fixed wrong-import-order and upgraded pip packages

> ./check/all 
> Running misc
> No issues.
> Running pylint
> ************* Module cirq.contrib.qasm_import._lexer_test
> cirq/contrib/qasm_import/_lexer_test.py:10:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.google.arg_func_langs_test
> cirq/google/arg_func_langs_test.py:19:0: C0411: third party import "from google.protobuf import json_format" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.google.common_serializers_test
> cirq/google/common_serializers_test.py:15:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.google.engine.engine
> cirq/google/engine/engine.py:42:0: C0411: third party import "import google.protobuf as gp" should be placed before "from apiclient import discovery, http as apiclient_http" (wrong-import-order)
> cirq/google/engine/engine.py:43:0: C0411: third party import "from google.protobuf import any_pb2" should be placed before "from apiclient import discovery, http as apiclient_http" (wrong-import-order)
> ************* Module cirq.google.optimizers.convert_to_sycamore_gates_test
> cirq/google/optimizers/convert_to_sycamore_gates_test.py:2:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.ion.convert_to_ion_gates_test
> cirq/ion/convert_to_ion_gates_test.py:17:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.ion.ion_device_test
> cirq/ion/ion_device_test.py:18:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.linalg.combinators_test
> cirq/linalg/combinators_test.py:15:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.linalg.transformations_test
> cirq/linalg/transformations_test.py:606:12: E1121: Too many positional arguments for method call (too-many-function-args)
> cirq/linalg/transformations_test.py:612:12: E1121: Too many positional arguments for method call (too-many-function-args)
> ************* Module cirq.neutral_atoms.convert_to_neutral_atom_gates_test
> cirq/neutral_atoms/convert_to_neutral_atom_gates_test.py:16:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.ops.arithmetic_operation_test
> cirq/ops/arithmetic_operation_test.py:16:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.ops.parallel_gate_operation_test
> cirq/ops/parallel_gate_operation_test.py:16:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.ops.pauli_interaction_gate_test
> cirq/ops/pauli_interaction_gate_test.py:17:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.ops.pauli_string_phasor_test
> cirq/ops/pauli_string_phasor_test.py:17:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.optimizers.convert_to_cz_and_single_gates_test
> cirq/optimizers/convert_to_cz_and_single_gates_test.py:17:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.optimizers.eject_z_test
> cirq/optimizers/eject_z_test.py:15:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.protocols.circuit_diagram_info_protocol
> cirq/protocols/circuit_diagram_info_protocol.py:19:0: C0411: third party import "from typing_extensions import Protocol" should be placed before "import sympy" (wrong-import-order)
> ************* Module cirq.protocols.json
> cirq/protocols/json.py:35:0: C0411: third party import "from typing_extensions import Protocol" should be placed before "import pandas as pd" (wrong-import-order)
> ************* Module cirq.protocols.json_test
> cirq/protocols/json_test.py:26:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.protocols.mixture_protocol_test
> cirq/protocols/mixture_protocol_test.py:17:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
> ************* Module cirq.sim.density_matrix_utils_test
> cirq/sim/density_matrix_utils_test.py:498:36: E1121: Too many positional arguments for method call (too-many-function-args)
> cirq/sim/density_matrix_utils_test.py:517:36: E1121: Too many positional arguments for method call (too-many-function-args)
> ************* Module cirq.sim.wave_function_test
> cirq/sim/wave_function_test.py:19:0: C0411: third party import "import numpy as np" should be placed before "import pytest" (wrong-import-order)
>

### CODE REVIEW

**Comment by chlorophyll-zz** [comment]
@googlebot I signed it!

File: .travis.yml [hunk 1]
```diff
@@ -9,33 +9,32 @@ branches:
 before_install:
 - pip install --upgrade pip setuptools wheel
 stages:
```
Comment (vtomole) [inline .travis.yml]: Wat's the reason for the changes in these file?

File: README.rst [hunk 2]
```diff
@@ -6,17 +6,10 @@
 Cirq is a Python library for writing, manipulating, and optimizing quantum
 circuits and running them against quantum computers and simulators.
```
Comment (vtomole) [inline README.rst]: What's the reason for these changes? Why did you re link the image to your repo?

File: README.rst [hunk 3]
```diff
@@ -6,17 +6,10 @@
...
 
-.. image:: https://travis-ci.com/quantumlib/Cirq.svg?token=7FwHBHqoxBzvgH51kThw&branch=master
-  :target: https://travis-ci.com/quantumlib/Cirq
+.. image:: https://travis-ci.com/chlorophyll-zz/Cirq.svg?branch=master
```
[the remark this answers is not in the document]
Comment (chlorophyll-zz) [reply README.rst]: i modified the travis.yml because the pylint and mypy tests were failing and therefore the builds were not tested. so i created a travis-ci instance to include the build tests anyway. 

i made the commit after the pull request. 
i did not kniow that this commit was also included in the pull request.
i guess i should have created another branch for that commit. 
i only wanted to include 
missing pylatex package 8063503a79fed3cf3eb4d491940d6bf842692685
and
mypy format errors 252b5fb7dc2e0ccf4bae86fb4bc06c50c1b21ae2
in the pull request. i am not that familiar with branches. i will learn github better and take tutorials.

File: .travis.yml [hunk 4]
```diff
@@ -9,33 +9,32 @@ branches:
...
-  - stage: quick_test
+  - stage: test
     os: linux
     env: NAME=format
     python: '3.6'
     install:
       - cat dev_tools/conf/pip-list-dev-tools.txt | grep yapf | xargs pip install
     script: check/format-incremental
-  - stage: quick_test
+  - stage: test
```
[the remark this answers is not in the document]
Comment (chlorophyll-zz) [reply .travis.yml]: i modified the travis.yml because the pylint and mypy tests were failing and therefore the builds were not tested. so i created a travis-ci instance to include the build tests anyway.
i did not kniow that this commit was also included in the pull request.

File: .travis.yml (hunk 1 above)
Comment (chlorophyll-zz) [reply .travis.yml]: i modified the travis.yml because the pylint and mypy tests were failing and therefore the builds were not tested. so i created a travis-ci instance to include the build tests anyway.
i did not kniow that this commit was also included in the pull request.

File: README.rst (hunk 2 above)
Comment (chlorophyll-zz) [reply README.rst]: i just wanted to have a quick link to the travis-ci build test. 
i did not wanted to include this patch in the pull request.

**Comment by vtomole** [comment]
@chlorophyll-zz Revert the changes we don't want in this branch and I'll merge it in.

File: dev_tools/auto_merge.py [hunk 5]
```diff
@@ -62,7 +62,7 @@ def from_github(repo: GithubRepository,
 
         if response.status_code != 200:
             raise RuntimeError(
-                'Pull check failed. Code: {}. Content: {}.'.format(
+                'Pull check failed. Code: {}. Content: {!r}.'.format(
```
Comment (Strilanc) [inline dev_tools/auto_merge.py]: Why is this change present? It's not related to what is described in the PR title or summary.

File: dev_tools/conf/pip-list-dev-tools.txt [hunk 6]
```diff
@@ -1,28 +1,27 @@
...
-mypy-protobuf==1.10
+grpcio-tools~=1.25
+mypy-protobuf==1.16
 
 # For uploading packages to pypi.
 twine
 
 # For verifying behavior of qasm output.
-qiskit~=0.13.0
+qiskit~=0.13
```
Comment (Strilanc) [inline dev_tools/conf/pip-list-dev-tools.txt]: This change is wrong. It means that if qiskit makes a breaking change our build will break.

File: dev_tools/conf/pip-list-dev-tools.txt [hunk 7]
```diff
@@ -1,28 +1,27 @@
...
 # For generating documentation.
 pypandoc
 recommonmark >= 0.4.0
 Sphinx
 sphinx_rtd_theme
 sphinx-markdown-tables
-
-# Need to pin pylint's parser to 2.1 instead of 2.2 until https://github.com/PyCQA/astroid/issues/650 is fixed
-astroid~=2.1.0
+pylatex
```
Comment (Strilanc) [inline dev_tools/conf/pip-list-dev-tools.txt]: Why was this dependency added?

File: dev_tools/conf/pip-list-dev-tools.txt [hunk 8]
```diff
@@ -1,28 +1,27 @@
...
+astroid~=2.3
```
Comment (Strilanc) [inline dev_tools/conf/pip-list-dev-tools.txt]: You can remove this instead of changing it, now that the bug forcing us to pin it is gone.

File: dev_tools/conf/pip-list-dev-tools.txt [hunk 9]
```diff
@@ -1,28 +1,27 @@
...
-pytest-cov~=2.5.0
-pytest-benchmark~=3.2.0
-yapf~=0.27.0
+mypy~=0.750
+pylint~=2.4
+pytest~=5.3
+pytest-asyncio~=0.10
+pytest-cov~=2.8
+pytest-benchmark~=3.2
+yapf~=0.29
```
Comment (Strilanc) [inline dev_tools/conf/pip-list-dev-tools.txt]: What kinds of changes to formatting does this cause?

File: dev_tools/conf/pip-list-dev-tools.txt [hunk 10]
```diff
@@ -1,28 +1,27 @@
 # For testing and analyzing code.
-mypy~=0.701.0
-pylint~=2.3.0
-pytest~=3.8.0
-pytest-asyncio~=0.10.0
-pytest-cov~=2.5.0
-pytest-benchmark~=3.2.0
-yapf~=0.27.0
+mypy~=0.750
```
Comment (Strilanc) [inline dev_tools/conf/pip-list-dev-tools.txt]: This change is wrong. It means that when mypy introduces new checks our build will break.

File: dev_tools/conf/pip-list-dev-tools.txt [hunk 11]
```diff
@@ -1,28 +1,27 @@
...
+pylint~=2.4
+pytest~=5.3
+pytest-asyncio~=0.10
```
Comment (Strilanc) [inline dev_tools/conf/pip-list-dev-tools.txt]: This change is wrong. It means that when pytest-asyncio introduces breaking changes our build will break.

File: .travis.yml [hunk 12]
```diff
@@ -9,33 +9,32 @@ branches:
 before_install:
```
Comment (Strilanc) [inline .travis.yml]: Revert all changes in .travis.yml.

File: requirements.txt [hunk 13]
```diff
@@ -1,14 +1,14 @@
...
 
 dataclasses; python_version < '3.7'
-google-api-python-client~=1.6
-matplotlib~=3.0
-networkx==2.3
-numpy~=1.16
+google-api-python-client~=1.7
+matplotlib~=3.1
+networkx==2.4
+numpy~=1.17
```
Comment (Strilanc) [inline requirements.txt]: Don't bump user version numbers in requirements.txt unless we actually *require* that version for the code to work. There should be some bug that was fixed or some feature we need. Otherwise we can create incompatibilities with other libraries e.g. if they require a specific version of numpy because the latest one is broken.

File: dev_tools/conf/pip-list-dev-tools.txt (hunk 9 above)
Comment (Strilanc) [reply dev_tools/conf/pip-list-dev-tools.txt]: Also, this change is wrong because it means that yapf making changes will break our build.

File: dev_tools/auto_merge.py (hunk 5 above)
Comment (chlorophyll-zz) [reply dev_tools/auto_merge.py]: fix pylint format warnings when running ./check/all

File: dev_tools/conf/pip-list-dev-tools.txt (hunk 10 above)
Comment (chlorophyll-zz) [reply dev_tools/conf/pip-list-dev-tools.txt]: do you want a == and exact version pin?

File: .travis.yml (hunk 12 above)
Comment (chlorophyll-zz) [reply .travis.yml]: yes

File: requirements.txt (hunk 13 above)
Comment (chlorophyll-zz) [reply requirements.txt]: incompatibilities were there before that prevented asteroids dependency from installing. 
the incompatibilities were coming from hard pinned specific versions. 
i have not documented that. 
i think i will split the pr in two branches, one for dependencies and one for pylint format fixes,

**Comment by vtomole** [comment]
I misspoke when I said I was going to merge this after the changes were reverted. There were just so many changes at once that we didn't care about. I didn't want to review it before they were fixed.

Feel free to open a cleaner pull request @chlorophyll-zz 

CC @Strilanc

**Comment by chlorophyll-zz** [comment]
@vtomole all right, no problem with that. i am getting better used to branches and pull request as i installed a python ide with great git support, pycharm community edition, you definately want to check that out. 
i can look what change fixed the asteroid bug. but i dont know of what asteroid bug you speak of. can you give me an error or a debug message, so i could narrow it down, say, to a package depency or a file that i patched?
